src/sentry/static/sentry/app/utils/tokenizeSearch.tsx

```diff
@@ -138,9 +138,11 @@ export class QueryResults {
     // to see if that open paren corresponds to a closed paren with one or fewer items inside.
     // If it does, delete those parens, and loop again until there are no more parens to delete.
     let parensToDelete: number[] = [];
-    const cleanParens = (_, idx) => !parensToDelete.includes(idx);
+    const cleanParens = (_, idx: number) => !parensToDelete.includes(idx);
     do {
-      this.tokens = this.tokens.filter(cleanParens);
+      if (parensToDelete.length) {
```

Comment:
Is this check necessary?

Reply:
It's an optimization. 

If `parensToDelete` is empty, then `cleanParens()` will always return `true.`

Reply:
I didn't see a point to filter a potentially long token list to do no work when there is nothing that could be removed. This removes a tiny bit of CPU burn.